### ISSUE TITLE
GQL-modules injection problem if no params to the function

### ISSUE THREAD

**Issue by mohamedGamalAbuGalala** [MichalLytek/type-graphql#588]
**Describe the bug**
I work on a project with `GQL-modules` and `type-graphql` both works together like a charm but I'm having some problems when I create a query/mutation and the function doesn't need to take any input, as it gives me an error that the injected instance is undefined.

**Module code** 
```javascript

export const HopeModule = new GraphQLModule({
  extraSchemas: [
    buildSchemaSync({
      resolvers: [HopeQuery],
      container: ({ context }) => context.injector
    })
  ],
  providers: [
    HopeQuery,
    HopeProvider
  ]
})

```

**Resolver code**
```javascript
   
@Resolver()
export class HopeQuery {
  constructor (private readonly hopeProvider: HopeProvider) {}

  @Query(() => String)
  hopeQuery (): string {
    return this.hopeProvider.init() // should return "I'm happy :)"
  }
}

```
**Expected behavior**
When I call the hopeQuery it returns the "I'm happy :)" string from the provider.

**Actual Behavior**
It gives me "Cannot read property 'init' of undefined" which means that hopeProvider is undefined.
But replacing this `hopeQuery` method with another taking any parameter works!!
```javascript
  @Query(() => String)
  hopeQuery (@Ctx() { me }: IMyContext): string {
    return this.hopeProvider.init() // should return "I'm happy :)"
  }
```
The last snippet will work!!

Any thoughts or ideas and if it not related to typegql, please told me so I can report that in the GQL-modules repo.

**Comment by mohamedGamalAbuGalala** [comment]
Urigo/graphql-modules#1019

**Comment by MichalLytek** [comment]
> Any thoughts or ideas and if it not related to typegql, please told me so I can report that in the GQL-modules repo.

For TypeGraphQL perspective, to resolve dependencies it only calls `container.get(HopeQuery)` and it's container responsibility to resolve and inject `HopeProvider`.

Placing additional decorators on resolver class methods arguments makes no difference to that 😕

**Comment by mohamedGamalAbuGalala** [comment]
After some debugging, I can say that it's not exactly as I reported, as it didn't work with arguments again for no reason. 
😞😞

**Comment by mohamedGamalAbuGalala** [comment]
Closing this after digging more in this lead me that the error was in some package or some piece of code affecting `reflect-metadata` package which blows everything.
But for referencing the error was in some changes in one file, when I discarded them it goes away.

Maybe related to connecting error with `neode` (related to adam-cowley/neode#56
 or adam-cowley/neode#14)or another error.

god-only-know-why the reason causing it.
